python-avd/pyavd/_eos_designs/structured_config/mlag/__init__.py

```diff
@@ -306,7 +306,13 @@ def router_bgp(self) -> dict | None:
             return None
 
         # MLAG Peer group
-        peer_group_name = self.inputs.bgp_peer_groups.mlag_ipv4_underlay_peer.name
+        if (
+            self.inputs.bgp_peer_groups.use_separate_mlag_peer_group_for_overlay
+            and self.inputs.bgp_peer_groups.mlag_ipv4_underlay_peer.name == "MLAG-IPv4-UNDERLAY-PEER"
+        ):
+            peer_group_name = "MLAG-IPv4-OVERLAY-PEER"
```

Comment:
this should not be changed here.
instead when this: `self.inputs.bgp_peer_groups.use_separate_mlag_peer_group_for_overlay` is true, the peer group used for MLAG overlay should be the one under `bgp_peer_groups.mlag_ipv4_overlay_peer`

Reply:
added!